File: kyonggiyo-core/src/main/java/kyonggiyo/application/auth/domain/vo/UserInfo.java
package kyonggiyo.application.auth.domain.vo;

import kyonggiyo.domain.user.Role;

public record UserInfo(
        Long userId,
        Role role
) {

    public static UserInfo from(AuthInfo authInfo) {
        return new UserInfo(authInfo.userId(), authInfo.role());
    }

}

File: kyonggiyo-core/src/main/java/kyonggiyo/application/auth/domain/vo/UserInfo.kt
package kyonggiyo.application.auth.domain.vo

import kyonggiyo.domain.user.Role

data class UserInfo(
        val userId: Long,
        val role: Role
) {

    companion object {
        fun from(authInfo: AuthInfo): UserInfo {
            return UserInfo(authInfo.userId, authInfo.role)
        }
    }

}
